File: frontend/src/types/augmentation.ts
export type AugTechniqueType = "flip" | "rotate" | "hsv" | "blur" | "noise";

export interface AugTechniqueConfig {
  type: AugTechniqueType;
  copies: number;
  angle_range: [number, number];
  hsv_h: number;
  hsv_s: number;
  hsv_v: number;
  blur_kernel_range: [number, number];
  noise_sigma_range: [number, number];
}

export interface AugmentRequest {
  source: string;
  destination: string;
  techniques: AugTechniqueConfig[];
}

export const AUG_TECHNIQUE_LABELS: Record<AugTechniqueType, string> = {
  flip: "Horizontal Flip",
  rotate: "Rotate",
  hsv: "HSV Jitter",
  blur: "Blur",
  noise: "Noise",
};

File: backend/app/schemas/augmentation.py
from typing import Literal

from pydantic import BaseModel

AugTechniqueType = Literal["flip", "rotate", "hsv", "blur", "noise"]


class AugTechniqueConfig(BaseModel):
    type: AugTechniqueType
    copies: int = 1
    angle_range: tuple[float, float] = (-15, 15)
    hsv_h: float = 0.015
    hsv_s: float = 0.7
    hsv_v: float = 0.4
    blur_kernel_range: tuple[int, int] = (3, 7)
    noise_sigma_range: tuple[float, float] = (5, 25)


class AugmentRequest(BaseModel):
    source: str
    destination: str
    techniques: list[AugTechniqueConfig]
